designs/Bignums-design.md

```diff
@@ -0,0 +1,104 @@
+# Compiler A: Bignums - design
+
+## AST, IR, and Built-in Libraries
+- We will change `print` in the `built-in library` to print bignums from the heap.
+- We plan to modify the `built-in library` to have Javascript functions support all the unary and binary operations of bignums. This way we can first support the behavior of Python, and then improve the efficiency by writing the functions in WASM.
+
+## New Functions, Datatypes, and/or Files
+- We will change `codeGenValue` to allocate bignums in the memory. This will store the bignum in memory and leave the address on the stack.
+- We will modify `codeGenUniOp` and `codeGenBinOp` to support operations.
+- We will add `codeGenMemoryIndex` to let strings or lists access memory with bignums as indices. This function will take the base address and the bignum index, and return the memory address you want to access.
```

Comment:
This is interesting – is the expectation that those groups would call this function somehow, or is there a new IR form for it? When does this get called?